wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -3,4 +3,106 @@
  * Add/update/retrieve Tapestry post and its child nodes
  * 
  */
-class TapestryController {}
\ No newline at end of file
+
+class TapestryController {
+    const POST_TYPES = [
+        'TAPESTRY' => 'tapestry',
+        'TAPESTRY_NODE' => 'tapestry_node'
+    ];
+    const ERRORS = [
+        'INVALID_POST_ID' => [
+            'MESSAGE' => 'PostID is invalid',
+            'STATUS' => ['status' => 404]
+        ]
+    ];
+    private $postId;
+
+    /**
+     * Constructor
+     */
+    public function __construct($postId) {
```

Comment:
This parameter should default to 0 since that will allow it to be initialized without a parameter in order to create a new tapestry. See "ID" parameter description for [wp_insert_post](https://developer.wordpress.org/reference/functions/wp_insert_post/#parameters).